### PR TITLE
task/DES-3025 Support OT data in recon portal through Tapipy

### PR DESCRIPTION
**Pull request by Beulah-Karrolla**
## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-3025](https://tacc-main.atlassian.net/browse/DES-3025)

## Summary of Changes: ##
Modified the opentopography data urls - `opentopodata_center/` & `opentopo_coordinates/`  to fetch community data using scripts through tapipy

## Testing Steps: ##
1.  Install tapipy using poetry
2.  make build
3.  Resolve dependencies(if any)
4. Confirm all services are running without errors.

## Notes: ##
- [Resolved] Some unit tests are failing with warnings, which are not related to this branch code. Not sure on how to proceed on that.

### CODE REVIEW

**Comment by Beulah-Karrolla** [comment]
Some unit tests are failing with warnings, which are not related to this branch code. Not sure on how to proceed on that.

File: designsafe/apps/rapid/views.py [hunk 1]
```diff
@@ -11,6 +13,8 @@
 from django.contrib.auth.decorators import login_required, user_passes_test
 from django.contrib.auth import get_user_model, models
 from django.db.models import Q
+from tapipy.tapis import Tapis
```
Comment (nathanfranklin) [inline designsafe/apps/rapid/views.py:16]: @Beulah-Karrolla , can you add the the changes to pyproject.toml and poetry.lock for the installed tapipy.

File: designsafe/apps/rapid/tests.py [hunk 2]
```diff
@@ -75,13 +75,13 @@ def test_get_opentopodata_center(self):
         self.assertContains(resp, "FeatureCollection")
 
     def test_get_opentopo_coordinates(self):
-        url = reverse('designsafe_rapid:get_opentopo_coordinates')
+        url = reverse('designsafe_rapid:get_opentopo_polygon_coordinates')
```
Comment (nathanfranklin) [inline designsafe/apps/rapid/tests.py:78]: we'll also need to update the tests by mocking the call via tapipy to get the file.  

after https://github.com/DesignSafe-CI/portal/pull/1373/files#r1688651134 is addressed, then the CI will fail on trying to get the file using tapis.   Now we can mock getting that file and return the example jsons we added previously to the repo.  

There could be different ways to mock the response from tapis.  We could mock TapiPy and the file.getContents method.  That might be the clearest way to do it. 🤔

File: designsafe/apps/rapid/views.py (hunk 1 above)
Comment (Beulah-Karrolla) [reply designsafe/apps/rapid/views.py:16]: addressed.

File: designsafe/apps/rapid/tests.py (hunk 2 above)
Comment (Beulah-Karrolla) [reply designsafe/apps/rapid/tests.py:78]: @nathanfranklin, for some clarification I am aware of mocking the files.getContents method to return the example json. 
But to mock Tapipy without auth details, can I use MagicMock or is there a preferred method

Comment (Beulah-Karrolla) [reply designsafe/apps/rapid/tests.py:95]: addressed.

**Review by fnets — APPROVED** [review]
LGTM and runs locally! Great work.

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍

**Pull request merged**